esp8266/modnetwork.c

```diff
@@ -113,9 +119,90 @@ STATIC mp_obj_t esp_isconnected() {
     }
     return mp_const_false;
 }
-
 STATIC MP_DEFINE_CONST_FUN_OBJ_0(esp_isconnected_obj, esp_isconnected);
 
+/// \method ifconfig(id = SPA|AP, config = [(ip, netmask, gateway, dns)])
+/// Return (ip, netmask, gateway, dns) if config is empty
+///     none otherwise.
+STATIC mp_obj_t esp_ifconfig(mp_uint_t n_args, const mp_obj_t *pos_args, mp_map_t *kw_args) {
+    struct ip_info ipconfig;
+
+    STATIC const mp_arg_t wlan_ifconfig_args[] = {
+        { MP_QSTR_id,               MP_ARG_INT,     {.u_int = 0} },
+        { MP_QSTR_config,           MP_ARG_OBJ,     {.u_obj = MP_OBJ_NULL} },
+    };
+
+    // parse args
+    mp_arg_val_t args[MP_ARRAY_SIZE(wlan_ifconfig_args)];
+
```

Comment:
It's nice to have readable code with right amount of whitespace, but this patch has it just too much, leaving an impression there's a blank line after each statement. MicroPython code has higher vertical code density and uses blank lines to separate long sequences of statements into integral blocks. It would be nice to follow the same codestyle.